src/utils/useProjectId.ts

```diff
@@ -0,0 +1,49 @@
+import { useChannel, useParameter } from "@storybook/manager-api";
+import React from "react";
+
+import { PROJECT_PARAM_KEY, PROJECT_UPDATED, UPDATE_PROJECT } from "../constants";
+
+let inMemoryProjectId: string | null = null;
+
+export const useProjectId = (): [
+  projectId: string,
+  updateProject: (projectId: string, projectToken?: string) => void,
+  projectIdChanged: boolean,
+  clearProjectIdChanged: () => void
+] => {
+  //
+  // Ideally this should only be configured once in main.ts. We'll need to figure out how to do that. This should at least read from that config for now.
+  const configuredProjectId = useParameter(PROJECT_PARAM_KEY, null);
```

Comment:
@ndelangen do you know of any way I could pull this configuration from the addon options in main.ts instead of preview.tsx (parameters)?

Our serverChannel methods seem to have access to options for projectToken, so I've left that value there. It seems like it might be better for our projectId to also be configured in `main.ts`.